### ISSUE TITLE
Flaky test `test_deterministic_of_observed_modified_interface`

### ISSUE THREAD

**Issue by ricardoV94** [pymc-devs/pymc#4661]
```
FAILED pymc3/tests/test_sampling.py::TestSamplePPC::test_deterministic_of_observed_modified_interface
```

This test failed here: https://github.com/pymc-devs/pymc3/runs/2423911879

And I think I have seen it before as well. We should check if it's a newly introduced bug or just needs seeding.

**Comment by ricardoV94** [comment]
Failed again here: https://github.com/pymc-devs/pymc3/pull/4646/checks?check_run_id=2512819254

**Comment by ricardoV94** [comment]
Appeared again in https://github.com/pymc-devs/pymc3/runs/2597322712

**Comment by ricardoV94** [comment]
Tentatively closed by #4707
